core/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java

```diff
@@ -93,4 +98,31 @@ public void execute(Runnable command) {
             }
         }
     }
+
+    @Override
+    public String toString() {
+        /*
+         * ThreadPoolExecutor has some nice information in its toString but we
+         * can't recreate it without nastier hacks than this.
+         */
+        String tpeToString = super.toString();
+        int startOfInfoInTpeToString = tpeToString.indexOf('[');
+        String tpeInfo;
+        if (startOfInfoInTpeToString >= 0) {
+            tpeInfo = tpeToString.substring(startOfInfoInTpeToString + 1);
+        } else {
+            assert false: "Unsupported ThreadPoolExecutor toString";
+            tpeInfo = tpeToString;
+        }
+        StringBuilder b = new StringBuilder();
+        b.append(getClass().getSimpleName()).append('[');
+        b.append(name).append(", ");
+        if (getQueue() instanceof SizeBlockingQueue) {
+            @SuppressWarnings("rawtypes")
+            SizeBlockingQueue queue = (SizeBlockingQueue) getQueue();
+            b.append("queue capacity = ").append(queue.capacity()).append(", ");
+        }
+        b.append("state = ").append(tpeInfo);
+        return b.toString();
+    }
```

Comment:
It'd look pretty bloated but otherwise return all the data. I think the tests would almost all pass as they are written too. It'd just be harder to read and funny looking.